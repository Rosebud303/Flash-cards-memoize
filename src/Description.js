import React from 'react';
import './_Description.scss'

export default function Description (props) {
    return(
        <div className='definition-card'>
            <div className='card'>
                <h1 className='back' onClick={props.toggle}>{props.proto.name}</h1>
                <p className='description'>{props.proto.definition}</p>
                <form>
                    <button className='answer' value='accessor' onClick={props.check}>Accessor</button>
                    <button className='answer' value='iteration' onClick={props.check}>Iteration</button>
                    <button className='answer' value='mutator' onClick={props.check}>Mutator</button>
                </form>
            </div>
        </div>
    )
}