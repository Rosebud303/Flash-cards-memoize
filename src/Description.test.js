import React from 'react';
import Description from './Description.js';
import { shallow } from 'enzyme';

const mockArray = [{
    "id": 1,
    "name": ".concat",
    "definition": "The concat() method is used to merge two or more arrays. This method does not change the existing arrays, but instead returns a new array.",
    "type": "accessor"
  },
  {
    "id": 2,
    "name": ".copyWithin",
    "definition": "The copyWithin() method shallow copies part of an array to another location in the same array and returns it, without modifying its size.",
    "type": "mutator"
  },
  {
    "id": 3,
    "name": ".entires",
    "definition": "The entries() method returns a new Array Iterator object that contains the key/value pairs for each index in the array.",
    "type": "iteration"
  }
]

const mockFunc = jest.fn()

describe('Description', () => {
    let wrapper;
    beforeEach(() => {
        wrapper = shallow (
            <Description proto={mockArray} 
                check={mockFunc}
            />
        )
    })

    it('should be a snapshot match', () => {
        expect(wrapper).toMatchSnapshot()
    })

    it('it should call the function being passed down', () => {
        wrapper.find('.back').simulate('click')
    })

    it('it should call the function being passed down', () => {
        wrapper.find('button').at(0).simulate('click')
    })

    it('it should call the function being passed down', () => {
        wrapper.find('button').at(1).simulate('click')
    })

    it('it should call the function being passed down', () => {
        wrapper.find('button').at(2).simulate('click')
    })
})