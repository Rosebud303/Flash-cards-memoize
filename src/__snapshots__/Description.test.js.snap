// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Description should be a snapshot match 1`] = `
<div
  className="definition-card"
>
  <div
    className="card"
  >
    <h1
      className="back"
    />
    <p
      className="description"
    />
    <form>
      <button
        className="answer"
        onClick={[MockFunction]}
        value="accessor"
      >
        Accessor
      </button>
      <button
        className="answer"
        onClick={[MockFunction]}
        value="iteration"
      >
        Iteration
      </button>
      <button
        className="answer"
        onClick={[MockFunction]}
        value="mutator"
      >
        Mutator
      </button>
    </form>
  </div>
</div>
`;
